app/waf/src/keys.ts

```diff
@@ -41,3 +43,20 @@ export namespace WafBindingKeys {
     );
   }
 }
+
+export function getTenantId(reqCxt: RequestContext): Promise<string> {
+  try {
+    let tenant_id = reqCxt.get(WafBindingKeys.Request.KeyTenantId);
+    return tenant_id;
+  } catch (error) {
+    // do nothing
+    throw error;
+  }
+}
+
+export function filterByTenantId<T extends CommonEntity>(
```

Comment:
This is a good abstraction of implementation, but I think it doesn't belong to keys.ts, I prefer you to move it to `waf/src/utils/tenant.ts` (`utils` is a new created folder, where we can put more common utils in it. In our code, there have been some utils code block to abstract).